src/main.rs

```diff
@@ -38,10 +41,9 @@ fn main() -> ! {
     // RCC = Reset and Clock Control
     let mut flash = dp.FLASH.constrain();
     let mut rcc = dp.RCC.constrain();
-    let clocks = rcc
-        .cfgr
-        .sysclk(32.mhz()) // Needed for SPI to work properly
-        .freeze(&mut flash.acr);
+    let clocks = rcc.cfgr.use_hse(8.mhz()).sysclk(48.mhz()).pclk1(24.mhz()).freeze(&mut flash.acr);
```

Comment:
a quick comment on what these numbers mean might be nice

Reply:
I added a few but it might not help much without reading the STM32 data sheets.